added files for debian packaging

unDocUMeantIt opened the pull request:
- the directory debian was changed, i.e., rewritten
  - it can be used for proper packaging, building three binary packages:
    libseafile, libseafile-dev (used for building the other seafile related
    packages), and seafile-cli (command line client)
  - this is what enabled the PPA https://launchpad.net/~m.eik/+archive/ubuntu/seafile-experimental

killing commented:
I have some ideas about package organization. I think only 5 packages are necessary:
- seafile-daemon: built from haiwen/seafile project. It contains the syncing engine and libseafile library. There is no need to built a separate libseafile and libseafile-dev package. The dev headers is of no use since I can't think of any third-party will be interested in it. We can added it when there is such requirements later.
- ccnet: built from haiwen/ccnet project. It contains the ccnet daemon and libccnet library. The current naming of libccnet is confusing. I suggest we change it to ccnet. It's a daemon, not a library.
- seafile-gui: built from haiwen/seafile-client project. Actually seafile-gui is a more proper name for the project.
- libsearpc: bulit from haiwen/libsearpc project.
- seafile-client: a meta-package to install the above 4 packages.

Additionally we also provides a seafile-cli package.

unDocUMeantIt commented:
thanks for sharing your ideas! you're right, the executables in libccnet need to get into a ccnet package, and adding a seafile-daemon package makes a lot of sense, as does naming the whole thing seafile-client. in effect, however, this will split things up in even more packages -- allthough that shouldn't be a problem.

for instance, libccnet contains the shared libraries that are linked with seafile-applet, so there should remain the libccnet package (and ccnet and seafile-gui will depend on it).

at first i thought that naming the meta-package different from the large official binary was a problem. but this could even be an improvement, as i can add "Conflicts: seafile" to the control file to make replacing one with the other much cleaner.

i've noticed i'll also need to rename the library packages to reflect their SONAME, and add a *.symbols file to them. i'll let you know when i'm ready.

unDocUMeantIt commented:
i've restructured the packaging:

- `libsearpc1` (shared libraries)
 - `libsearpc-dev` (development files to build with shared libs)
- `ccnet` (ccnet executables)
 - `libccnet0` (shared libraries)
 - `libccnet-dev` (development files to build with shared libs)
- `seafile-daemon` (the seafile daemon)
 - `libseafile0` (shared libraries)
 - `libseafile-dev` (development files to build with shared libs)
 - `seafile-cli` (comand line interface)
- `seafile-gui` (seafile desktop client)

i did not add another package called `seafile-client` because i noticed that `seafile-gui` does already fulfill the task. i.e., if you install `seafile-gui`, all necessary dependencies (including `libsearpc1`, `ccnet` and `seafile-daemon`) will be installed as well.

also, i did not yet add symbols files to the library packages. but what do you think so far?

i've set up a new, clean PPA with this packaging scheme, you can try/examine it here: 

- https://launchpad.net/~m.eik/+archive/ubuntu/seafile

last but not least, here's an updated script that temporarily clones the four github repos i forked and prepares ready-to-build debian source packages in the specified directory:

- http://reaktanz.de/stuff/oathaiqu/deb_git_seafile.sh

i used that in conjunction with `backportpackage` (from package `ubuntu-dev-tools`) to upload the package sources to the PPA.

unDocUMeantIt commented:
i've now added initial symbols files to packages with shared libraries, and fixed various issues lintian was complaining about. the packages are in pretty good shape already.

the remaining issues include an error regarding the installation target of the pkg-config files i don't fully grasp yet (regarding /usr/lib/pkg-config and multiarch packages), missing documentation/manpages for several executables (which unfortunately i can't write), and complaints about python scripts that are installed to /usr/bin having the *.py extension. those should be renamed (no extension) if possible, to comply with debian packaging standards.

lins05 commented on debian/control:
```diff
@@ -1,47 +1,74 @@
+    valac,
+    libjansson-dev,
+    python,
+    libfuse-dev,
```
Please remove the deps on `libfuse-dev`, and add `--disable-fuse` in the configure flags. 

It is a legacy problem: fuse is only used in seafile server, and should be automatically disabled in `configure.ac`. I'll fix that later, but for now let's use `--disable-fuse` flag.

unDocUMeantIt replied:
done.

lins05 commented on debian/copyright:
```diff
@@ -1,18 +1,44 @@
-is licensed under the GPL, see `/usr/share/common-licenses/GPL'.
+Files: debian/*
+Copyright: 2016 m.eik michalke
+License: GPL-3
```
Could you please change it from `GPL-3` to `GPL-2`? We have some legacy code borrowed from git, which we plan to remove in the near future so we can make the whole project to use GPLv3, but until then it's better to keep the whole project licensed under GPLv2.

The same for the your other PRs (https://github.com/haiwen/libsearpc/pull/26 , https://github.com/haiwen/ccnet/pull/109 , and https://github.com/haiwen/seafile-client/pull/807).

unDocUMeantIt replied:
done.

lins05 commented on debian/libseafile0.symbols:
```diff
@@ -0,0 +1,430 @@
+ seafile_web_access_set_op@Base 5.1.2
+ seafile_web_access_set_repo_id@Base 5.1.2
+ seafile_web_access_set_username@Base 5.1.2
+ seafile_web_query_access_token@Base 5.1.2
```
Could we generate the symbols file at build time instead of keeping this static list in the source tree?

unDocUMeantIt replied:
as far as i understand this part of the packageing process, technically, it would be possible, but in practice it would contradict the purpose of the file.

when other packages link this library, the symbols file can be queried to calculate the actual dependencies automatically (depending on what symbols are actually used). i.e., the version numbers you see in this file should only be changed if newer versions introduce incompatible changes. you should see a diff during the build of new versions if the included symbols have changed, and then use the updated symbols file from the newly build package. here's more details on the subject:
https://www.debian.org/doc/manuals/maint-guide/advanced.en.html#librarysymbols

lins05 replied:
SGTM. I'll read about it, thanks for the explanation!

lins05 commented:
@unDocUMeantIt Since there is some confliction between master and your branch, I squashed all the commits and pushed it to master. See https://github.com/haiwen/seafile/commit/dd20ca98155221dc269e0f2157d2f880400aca5f .

Thanks for the PRs!

eDeviser commented:
Hello and thank you for creating a ppa for seafile client! I do really enjoy it! :-)

Is it possible to add an deb for arm?

This would be awesome, because there are many people out there,  using Ubuntu Touch System on their smartphones or tablets who could then use seafile client, too.
